### ISSUE TITLE
differences between illumina sequencing analysis viewer and interop python package 

### ISSUE THREAD

**Issue by khl0798** [Illumina/interop#185]
I downloaded the illumina sequencing analysis viewer (version：1.8.37) on my local windows 10 laptop and  interop python package （version：1.1.4 ) on centos6.4  server(python 2.7)。I took the nextseqCN500 data as example.
Howerer,  Run Q30 data by interop package was conflicted with illumina sequencing analysis viewer client.
**### The differences were shown as follows：**
1) illumina sequencing analysis viewer：
![illuminasequencingnextseqcn500](https://user-images.githubusercontent.com/7838505/46593261-0f7a2580-cafd-11e8-99c7-668fc3f89ad6.png)
2)interop python package:
![interoppythonnextseqcn500](https://user-images.githubusercontent.com/7838505/46593268-1a34ba80-cafd-11e8-98cf-e0a9780d9b14.png)
**### Data:**
[180906.zip](https://github.com/Illumina/interop/files/2454752/180906.zip)

**### interop python package ran on centos 6.4 server：**
[@SAA-Server ~]$ cat /proc/version 
Linux version 2.6.32-358.el6.x86_64 (mockbuild@x86-022.build.eng.bos.redhat.com) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-3) (GCC) ) #1 SMP Tue Jan 29 11:47:41 EST 2013

**### illumina sequencing analysis viewer ran on windows 10 client:**
![windows10system](https://user-images.githubusercontent.com/7838505/46593489-f70b0a80-cafe-11e8-8b0f-65c7840080f0.png)

**### code:**
```import os
import pytz
import dateutil
import pandas as pd
import csv
import math
import argparse
import re
import glob


def parse_args():
    parser = argparse.ArgumentParser()
    parser.add_argument('-path', '--illumina_sequence_bin_data', required=True)
    parser.add_argument('-o', '--outpath', required=True, )
    parser.add_argument('-s', '--sample_name', required=True)
    args = parser.parse_args()
    interop_path = args.illumina_sequence_bin_data
    outpath = os.path.abspath(args.outpath)
    sample_name = args.sample_name
    return interop_path, outpath, sample_name


def format_value(val):
    if hasattr(val, 'mean'):
        return val.mean()
    else:
        return val


def format_value_median(val):
    if hasattr(val, 'median'):
        return val.median()
    else:
        return val


def rmnan(input):
    if math.isnan(input):
        return ""
    else:
        return input


def surface(summary, input=None, read_index=0, lane_index=0, check_type=None):
    if check_type == "tile_count":
        return format_value(getattr(summary.at(read_index).at(lane_index), "tile_count")())
    elif check_type == "error_rate":
        return format_value(getattr(summary.at(read_index).at(lane_index), "error_rate")())
    elif check_type == "error_rate_75":
        return format_value(getattr(summary.at(read_index).at(lane_index), "error_rate_75")())
    elif check_type == "error_rate_100":
        return format_value(getattr(summary.at(read_index).at(lane_index), "error_rate_100")())
    elif check_type == "error_rate_35":
        return format_value(getattr(summary.at(read_index).at(lane_index), "error_rate_35")())
    elif check_type == "percent_gt_q30":
        return format_value_median(getattr(summary.at(read_index).at(lane_index), "percent_gt_q30")())
    elif check_type == "density":
        return round(format_value(getattr(summary.at(read_index).at(lane_index), "density")()) / (1e+3))
    elif check_type == "percent_pf":
        return round(format_value(getattr(summary.at(read_index).at(lane_index), "percent_pf")()), 2)
    elif check_type == "density_pf":
        return format_value(getattr(summary.at(read_index).at(lane_index), "density_pf")())
    elif check_type == "phasing":
        return format_value_median(getattr(summary.at(read_index).at(lane_index), "phasing")())
    elif check_type == "cluster_count_pf":
        return format_value(getattr(summary.at(read_index).at(lane_index), "cluster_count_pf")())
    elif check_type == "prephasing":
        return format_value(getattr(summary.at(read_index).at(lane_index), "prephasing")())
    elif check_type == "first_cycle_intensity":
        return format_value(getattr(summary.at(read_index).at(lane_index), "first_cycle_intensity")())
    elif check_type == "yield_g":
        return round(format_value(getattr(summary.at(read_index).at(lane_index), "yield_g")()), 1)
    elif check_type == "reads":
        return round(format_value(getattr(summary.at(read_index).at(lane_index), 'reads')()) / (1e+6), 2)
    elif check_type == "reads_pf":
        return round(format_value(getattr(summary.at(read_index).at(lane_index), "reads_pf")()) / (1e+6), 2)
    elif check_type == "percent_aligned":
        return format_value(getattr(summary.at(read_index).at(lane_index), "percent_aligned")())
    elif check_type == "lane":
        return format_value(getattr(summary.at(read_index).at(lane_index), "lane")())
    else:
        return ""


# error_rate_read_lane_surface = numpy.zeros((summary.size(), summary.lane_count()))
def interopTable(input, outpath=None, sample_name=None, out=False):
    return_info = {}
    if out:
        output = os.path.join(outpath, sample_name + ".csv")
    run_metrics = py_interop_run_metrics.run_metrics()
    summary = py_interop_summary.run_summary()
    valid_to_load = py_interop_run.uchar_vector(py_interop_run.MetricCount, 0)
    py_interop_run_metrics.list_summary_metrics_to_load(valid_to_load)
    run_folder_path = input
    try:
        run_metrics.read(run_folder_path, valid_to_load)
    except Exception as ex:
        print("Skipping - cannot read RunInfo.xml: %s - %s" % (run_folder_path, str(ex)))
    py_interop_summary.summarize_run_metrics(run_metrics, summary)
    # print(summary.size(), summary.lane_count(), summary.surface_count())
    columns = (
        ('Yield Total (G)', "yield_g"), ('Projected Yield (G)', "projected_yield_g"), ("Error Rate", "error_rate"),
        ('% Aligned', "percent_aligned"), ("Intensity Cycle 1", "first_cycle_intensity"), ("%>=Q30", "percent_gt_q30"))
    read_lengthrows = [
        ("Read %s%d" % ("(I)" if summary.at(i).read().is_index() else " ", summary.at(i).read().number()),
         summary.at(i).summary()) for i in range(summary.size())]
    read_title = [i[0] for i in read_lengthrows]
    summary_title = pd.DataFrame({"Run Summary": []})
    if out:
        summary_title.to_csv(output, index=0, header=1)
    # error_rate first_cycle_intensity percent_aligned percent_gt_q30 projected_yield_g yield_g
    rows = [('Total', summary.total_summary())]
    d = []
    for label, func in columns:
        d.append((label, pd.Series([getattr(r[1], func)() for r in rows], index=[r[0] for r in rows])))
    rows = [("Read %s%d" % ("(I)" if summary.at(i).read().is_index() else " ", summary.at(i).read().number()),
             summary.at(i).summary()) for i in range(summary.size())]
    d_read = [] 
    for label, func in columns:
        d_read.append((label, pd.Series([getattr(r[1], func)() for r in rows], index=[r[0] for r in rows])))
    columns_list = ['Yield Total (G)', 'Projected Total Yield (G)', '% Aligned', "Error Rate",
                    "Intensity Cycle 1", "%>=Q30"]
    df_read = pd.DataFrame.from_items(d_read)
    df = pd.DataFrame.from_items(d)  # total
    d_read.append(d)  
    return_summary_data = []
    for index in range(len(df_read.values)):
        tmp = {}
        tmp["Level"] = read_title[index]
        for index_index in range(len(df_read.values[index])):
            if math.isnan(df_read.values[index][index_index]):
                tmp[columns_list[index_index]] = ""
            else:
                tmp[columns_list[index_index]] = df_read.values[index][index_index]
        return_summary_data.append(tmp)
    tmp = {}
    tmp["Level"] = "Total"
    result = pd.concat([df_read, df])
    for index_index in range(len(df.values[0])):
        if math.isnan(df.values[0][index_index]):
            tmp[columns_list[index_index]] = ""
        else:
            tmp[columns_list[index_index]] = df.values[0][index_index]
    return_summary_data.append(tmp)
    if out:
        result.to_csv(output, mode='a')
    columns_read = (
        ("Lane", "lane"), ('Tiles', 'tile_count'), ("Density (K/mm2)", 'density'), ("Cluster PF(%)", "percent_pf"),
        ("Phas/Prephas", "phasing"),
        ("Reads(M)", "reads"), ("Reads PF(M)", "reads_pf"), ("%>=Q30", "percent_gt_q30"),
        ("Yield(G)", "yield_g"), ("Aligned(%)", "percent_aligned"), ("Error Rate", "error_rate"),
        ("Error Rate 35 cycle(%)", "error_rate_35"), ("Error Rate 75 cycle(%)", "error_rate_75"),
        ("Error Rate 100 cycle(%)", "error_rate_100"), ("Intensity Cycle 1", "first_cycle_intensity"),
        ("percent_pf", "percent_pf"),
        ("cluster_count_pf", "cluster_count_pf"), ("prephasing", "prephasing"))
    ###summary.lane_count()  summary.read()  summary.
    different_read = []
    return_read_data = []
    for read_id in range(len(read_title)):
        read_title_write = pd.DataFrame({read_title[read_id]: []})
        if out:
            read_title_write.to_csv(output, mode='a', header=2, index=0)
        d_read = []
        tmp = {}
        tmp["data"] = []
        tmp["title"] = read_title[read_id]
        for lane_id in range(summary.lane_count()):
            lane_read = []
            for label, func in columns_read:
                add_info = pd.Series(surface(summary, read_index=read_id, lane_index=lane_id, check_type=func))
                if label == 'Tiles':
                    lane_read.append((label, 2 * float(add_info)))
                else:
                    lane_read.append((label, add_info))
            lane_read_tmp = pd.DataFrame.from_items(lane_read)
            d_read.append(lane_read_tmp)
        result = pd.concat(d_read)
        for index in range(len(result.values)):
            tmp_tmp = {}
            for index_index in range(len(result.values[index])):
                if math.isnan(result.values[index][index_index]):
                    tmp_tmp[columns_read[index_index][0]] = ""
                else:
                    tmp_tmp[columns_read[index_index][0]] = result.values[index][index_index]
            tmp['data'].append(tmp_tmp)
        return_read_data.append(tmp)
        different_read.append(result)
        if out:
            result.to_csv(output, mode='a', header=True, index=False)
    result_read = pd.concat(different_read)
    info_data = {"summary": return_summary_data, "readData": return_read_data}
    return_info["success"] = True
    return_info["info"] = info_data
    return return_info```

**Comment by ezralanglois** [comment]
What version of SAV are you comparing against?

**Comment by nudpa** [comment]
Hello @khl0798, thanks for the detailed description. The root cause of the discrepancy is that SAV 1.8.37 is somewhat old - the InterOp Python package is consistent with the latest SAV (2.4.5 here: http://emea.support.illumina.com/sequencing/sequencing_software/sequencing_analysis_viewer_sav/downloads.htm).

Specifically in this case, we learned that the last cycle of every read is automatically trimmed by many alignment pipelines due to its low quality. In order to get a more realistic expectation of the downstream data quality using these pipelines, we removed that cycle from summary statistics such as yield and %Q30. This was implemented in SAV 1.11.1 and later.
